### ISSUE TITLE
[Shaders] Very low FPS with Minecraft 1.16.1 shaders

### ISSUE THREAD

**Issue by BALDO3000** [sp614x/optifine#4288]
I have a Nvidia GTX 1660 ti and I usually get 100+ fps with shaders in the overworld. In Minecraft 1.16.1 my fps are really low, around 30-60 fps in the overworld. I tried to change the shaders pack but nothing changed, even lowering the shaders' quality.
I'm using OptiFine 1.16.1 HD U G2 pre2 installed with the included installer with no mods.

**Comment by willyde33** [comment]
I've had that before, a possible fix is just reset your minecraft video settings.. might be in extras? and start fresh. it may or may not help but it did for me and it doesnt take long to go through settings.

**Comment by Kewun0** [comment]
Same here, gtx 1650 super.

**Comment by GarethPW** [comment]
I've noticed similar performance using both intensive shaders and light ones, so I suspect there must be a bottleneck somewhere.

**Comment by Durjam** [comment]
Yeah i was looking at my gpu usage and fps when i play, and with internal shaders i get 100-200fps, with about 12% gpu usage, but when i usage shaders i get like 30-50% gpu usage with like 50fps (my computer is good - rtx2070s and ryzen 7 2700x) it seems like minecraft isnt utilizing my gpu enough to reach the framerates it should be capable of

**Comment by kevorino** [comment]
Same here, I've been spending too much time trying to figure out if this is OptiFine, JAVA or Minecraft in general... 

My specs:
Ryzen 7 3700x (Game Boost Enabled)
GTX 1080Ti
32GB RAM (12 or 16 for MC)
Vanilla 1.16.1 MC with OptiFine
Shaders Enabled
Usually high in-game settings, 16 render distance, Fancy graphics, high mipmaps, all particles.. etc...
FPS only goes higher than 50fps if my render distance is lower than 4.

I've tried so many shaders and every single one starts of with very low fps.
For example, Suldurs Vanilla Shaders, no matter what settings I use for shaders it's really low fps. If I set everything to the lowest (and I mean everything off and/or low), it stays at no more than 50 fps.

As a comparison, my favorite shaders is KUDA Shaders, in previous versions of the game and OptiFine I used the exact same KUDA Shaders and settings and got to 100+ FPS. Surprisingly, these precious versions were played on a Ryzen 5 1600x and lower RAM on MC instead of my current CPU. (Ryzen 7 3700x)

**Comment by thomashavard** [comment]
Hi. I'm using a gtx 1650 super and was getting roughly 10 fps with the furthest render distance settings. I switched the render distance to 16 chunks and the fps improved significantly to around 90+. Im using BSL shaders on 1.16.2.

**Comment by Beksim** [comment]
@thomashavard Yeah that checks out. 64 render distance is an insane amount of stuff to keep loaded at all times. And to top that off while using BSL.... that means the shader has to affect that whole distance. Your gpu cant handle that.

 @kevorino Sounds to me like you are allocating way too much ram to MC. More =/= better. I have a 2700x 2080 and 32gb ram as well and i usually only allocate 2-4 gbs for just vanilla with Ray traced shaders.

@Durjam Internal shaders arent meant to be played with. all it does is load the shader framework. use an actual shader or no shaders. regardless 50fps with a high tier shader on a 2070 sounds about right. you said plural 2070s. im assuming task manager is combining both of them so really the one mc uses is running at 60-100% I only have one 2080 so not 100% sure on if task manager combines them.

**Comment by francoisM77** [comment]
> @thomashavard Yeah that checks out. 64 render distance is an insane amount of stuff to keep loaded at all times. And to top that off while using BSL.... that means the shader has to affect that whole distance. Your gpu cant handle that.
> 
> @kevorino Sounds to me like you are allocating way too much ram to MC. More =/= better. I have a 2700x 2080 and 32gb ram as well and i usually only allocate 2-4 gbs for just vanilla with Ray traced shaders.
> 
> @Durjam Internal shaders arent meant to be played with. all it does is load the shader framework. use an actual shader or no shaders. regardless 50fps with a high tier shader on a 2070 sounds about right. you said plural 2070s. im assuming task manager is combining both of them so really the one mc uses is running at 60-100% I only have one 2080 so not 100% sure on if task manager combines them.

I have a the same card as Durjam, a 2070s, I think you got slightly confused and thought he meant a plural of 2 2070 cards, when it is the abbreviation for 2070 super. I get awful fps on sildurs, SEUS and BSL shaders with around 9-15 and occasional 0 lag spikes and have no idea whats causing it. Without my 64x64 texture pack i get around 20.

**Comment by SuspiciousLookingOwl** [comment]
Might be helpful for someone:

1. **Make sure Minecraft (Java) is using your discrete GPU**, not integrated. you can check this by opening `Shaders` in options and check on the bottom of the screen:
![image](https://user-images.githubusercontent.com/32597776/93719235-5444dd00-fbab-11ea-801c-5702062fd7d5.png)
If it says your discrete GPU (Nvidia / AMD), that means it's already using your discrete GPU. 

If it's using integrated GPU (Intel HD or AMD Vega), change your GPU settings, follow [this guide](https://windowsreport.com/minecraft-not-using-gpu/#:~:text=Click%20the%20Switchable%20Graphics%20tab,option%20for%20the%20discrete%20GPU.) to change what GPU to use in Minecraft (guide for both AMD and Nvidia card are there)

2. **Reduce render distance**, I found that some shaders are poorly optimized when you have a high render distance (8+), some that I know are BSL and Sildurs, I usually use 8-10 render distance with BSL / Sildurs, more than that I will start getting FPS drop. Some shaders doesn't impact too much performance when using high render distance, even on 48 chunks, like SEUS Renewed.

**Comment by VincyZed** [comment]
Same here, on pretty much every shader pack I have tried. Stays at 30-45 FPS and I have a 3700X and a GTX 1080. Playing on ultra-wide 1080p.

**Comment by KennethLim314** [comment]
I can verify this.

3700X, 2070, 32GB ram and 
On 1.12.2, I observe full GPU utilization and maxed (144) fps with shaders on (Sildurs medium)
in 1.16.3, with only optifine and shaders agin, I see 70% GPU utilization and unmaxed FPS.

In actually built up worlds, this difference is excaberated, with FPS dropping to sub-40 and GPU utilization dropping to ~30-40% as well.

This implies very strongly to me that we're observing heavy CPU bottlenecking along the way.

**Comment by kevorino** [comment]
> Same here, I've been spending too much time trying to figure out if this is OptiFine, JAVA or Minecraft in general...
> 
> My specs:
> Ryzen 7 3700x (Game Boost Enabled)
> GTX 1080Ti
> 32GB RAM (12 or 16 for MC)
> Vanilla 1.16.1 MC with OptiFine
> Shaders Enabled
> Usually high in-game settings, 16 render distance, Fancy graphics, high mipmaps, all particles.. etc...
> FPS only goes higher than 50fps if my render distance is lower than 4.
> 
> I've tried so many shaders and every single one starts of with very low fps.
> For example, Suldurs Vanilla Shaders, no matter what settings I use for shaders it's really low fps. If I set everything to the lowest (and I mean everything off and/or low), it stays at no more than 50 fps.
> 
> As a comparison, my favorite shaders is KUDA Shaders, in previous versions of the game and OptiFine I used the exact same KUDA Shaders and settings and got to 100+ FPS. Surprisingly, these precious versions were played on a Ryzen 5 1600x and lower RAM on MC instead of my current CPU. (Ryzen 7 3700x)

Alright, so results to my issue quoted here...

I started getting crashes on my PC after a while (30 minutes or less) of playing MC 1.16.3 with OptiFine and shaders ALONG WITH the constant non-sense lag.
After many updates, re-installs, edits, I made up my mind and got the money together to upgrade my components. (Start trying to figure out if it's a component issue or windows)
I went ahead and bought a new motherboard, CPU closed loop water cooler and got a fresh install of Windows 10 Home. I'm guessing the issue might have been my motherboard or windows install because... it is fixed as of "now". ( I've played 1 hour and no low frame rate or crashes) I even have even tried with CPU boost on, RAM overclock, other settings that I mention below that I usually change will not be messed with this time... For now...

My frame rate is way higher than it used to be, I am even able to crank up my video setting a a bit and get nice FPS.
I honestly don't know what exactly might have been the issue but I can provide that I tried almost everything possible without having to delete your windows version and installing a new fresh version.

**START OF THINGS I DID/DO TO MY PC**

GPU Driver Re-install
Mobo driver updates (BIOS and CPU)
Minecraft reinstall
OptiFine reinstall
Turn down graphics settings
Change settings in nvidia control panel
Disable cpu boost (small overclock provided in bios)
Disable manual RAM overclock
Enable the above options ^
A L O T of stuff

I can say what I do/did to my pc in general, I believed that giving specific programs like Minecraft or Discord the "Open as Administrator" option would improve performance so I did that with some programs.

I remember going into task manager and setting the "priority of resources" (I think it was called) to a higher priority on programs like Minecraft.

I tend to max out my Vanilla Minecraft settings (besides render distance) and play. (This usually plays well without shaders but I remember doing it and getting good frames, which I now do after upgrade)

I always personalize my windows settings, dark theme, appearance is set to max performance in advanced system settings, xbox game mode is disabled and all of its services, allow no apps to run in the background, disable almost all start up apps besides audio software and stuff that's needed

I also always allocated MC anywhere from 8 to 16 gigs or ram. (I have 32)
On updated version ive been sticking to 8GB

I used geforce experience (and still do/did) to record gameplay on the "clip" constant 2 minute loop. @ 1080p60 and 50mbps. On this I disabled and deleted all keybinds and extra features, highlights, streaming, notifications, whatever...

I started using discord overlay in Minecraft, it worked before and I still use it and it works

I always cleaned my pc, so there shouldn't have been thermal problems...

**END OF THINGS I DID TO MY PC**

And I can't remember other things...anyways, hopefully this helps some of y'all out there.

More testing to come and more updated to this..thread?.. I will post my old and new PC specs below:
(Post your questions or if you would like for me to check on something, my stats or something)

Add on discord: kev#9817

Updated PC:
https://pcpartpicker.com/user/keV_/saved/4mmjZL

Old PC:
https://pcpartpicker.com/list/msQCy4

**Comment by kevorino** [comment]
^ Check the thread in which I mentioned this issue for a small update before the following update. ^

**Update:**
Issue persists with specific shaders under high (but not abnormal) loads, I believe its a GPU related issue.

I have GTX1080ti, I have tried RTX2070Super and gameplay felt smoother + no crashing. The 2070Super and 1080ti are relatively similar on performance.

I have ran multiple GPU benchmarks and stress tests and no crashes or "freezes".

Leads me to believe that my GPU has a fault or there is a issue with compatibility between game with shaders and GPU??

ugh...have to wait for 3080 to be available now...goodbye until then?

**Comment by hazardousedens** [comment]
> Might be helpful for someone:
> 
> 1. **Make sure Minecraft (Java) is using your discrete GPU**, not integrated. you can check this by opening `Shaders` in options and check on the bottom of the screen:
>    ![image](https://user-images.githubusercontent.com/32597776/93719235-5444dd00-fbab-11ea-801c-5702062fd7d5.png)
>    If it says your discrete GPU (Nvidia / AMD), that means it's already using your discrete GPU.
>    If it's using integrated GPU (Intel HD or AMD Vega), change your GPU settings, follow [this guide](https://windowsreport.com/minecraft-not-using-gpu/#:~:text=Click%20the%20Switchable%20Graphics%20tab,option%20for%20the%20discrete%20GPU.) to change what GPU to use in Minecraft (guide for both AMD and Nvidia card are there)
> 2. **Reduce render distance**, I found that some shaders are poorly optimized when you have a high render distance (16+), some that I know are BSL and Sildurs, I usually use 8-12 render distance with BSL / Sildurs, more than that I will start getting FPS drop especially when I already rendered all of the chunks within the render distance area. Some shaders doesn't impact too much performance when using high render distance, even on 48 chunks, like SEUS Renewed.
>    For performance reference (I'm using Ryzen 3600 & GTX 1080ti on 1440p monitor):
>    
>    * **BSL v7.2**
>      8 chunks = 90-100 fps
>      16 chunks = 80-90 fps ( **-10%** from 8 )
>      32 chunks = 55-65 fps ( **-37%** from 8, **-29%** from 16)
>    * **Sildurs v1.262 Extreme VL**
>      8 chunks = 115-120 fps
>      16 chunks = 104-110 fps ( **-8.5%** from 8 )
>      32 chunks = 65-70 fps ( **-42%** from 8, **-36.5%** from 16)
>    * **Seus Renewed v1.0.1**
>      8 chunks = 100-105 fps
>      16 chunks = 90-95 fps ( **-9.8%** from 8 )
>      32 chunks = 89-92 fps ( **-11.8%** from 8, **-3.23%** from 16)

Thank you so much for sharing the GPU fix. I was getting ~60fps vanilla and 5fps with shaders on my high-end laptop and didn't know what was wrong. Now I get 250+fps vanilla and 80-100fps with shaders!

**Comment by brannez** [comment]
Okay, so I was having the same problem with shaders. I was getting nearly identical FPS on ultra shader packs and lite shader packs which is not normal. Turns out that I had biome blend on at 15x which doesn't really affect the look of the game that much when shaders are on. It made my FPS for both packs skyrocket to 140 FPS which was great. Biome blend is only really a useful aesthetic feature on vanilla MC.

**Comment by sp614x** [comment]
Should there be a warning for Biome Blend above 5?

**Comment by brannez** [comment]
> Should there be a warning for Biome Blend above 5?

I would say so.. it does say that performance could be impacted. But it definitely does impact on performance with shaders enabled. It doesn't impact performance when you are not using shaders. Not sure why that it is, just what I have found. But, definitely do think that a warning should be added.

**Comment by yermolim** [comment]
Same thing. Have around 10-20 fps with any shaders and any shader presets (from low to ultra), any resolution (from tiny window to 4k) and any draw distance on my i5-8400 + gtx1080ti. GPU load is around 30-70% depending on settings. Game version is 1.16.3. Optifine version is G5.

**Comment by kevorino** [comment]
> ^ Check the thread in which I mentioned this issue for a small update before the following update. ^
> 
> **Update:**
> Issue persists with specific shaders under high (but not abnormal) loads, I believe its a GPU related issue.
> 
> I have GTX1080ti, I have tried RTX2070Super and gameplay felt smoother + no crashing. The 2070Super and 1080ti are relatively similar on performance.
> 
> I have ran multiple GPU benchmarks and stress tests and no crashes or "freezes".
> 
> Leads me to believe that my GPU has a fault or there is a issue with compatibility between game with shaders and GPU??
> 
> ugh...have to wait for 3080 to be available now...goodbye until then?

**Conclusion for my issue**
GPU was the problem, Zotac AMP GTX 1080Ti
The card was purchased as USED a while back and not until recently, I learned that the card was being used for bit coin mining. This is not good.. it seems like the card had intermident issues and is what was causing my frames to drop and my game to crash.
I now have a new rtx 3080 and everything runs perfectly fine. Ugh, stupid graphics card.

**Comment by VincyZed** [comment]
Biome Blend hasn't improved things a lot on my end, maybe 10 fps but that's about it.

**Comment by yermolim** [comment]
> Same thing. Have around 10-20 fps with any shaders and any shader presets (from low to ultra), any resolution (from tiny window to 4k) and any draw distance on my i5-8400 + gtx1080ti. GPU load is around 30-70% depending on settings. Game version is 1.16.3. Optifine version is G5.

Tried again without any mods (only OF with default settings) - same thing around 20 fps with shaders regardless of settings. TPS is stable. Lagometer is mostly white. CPU load is around 30% (none of the cores hits 100%), GPU load is around 35%. Single or multiplayer - doesn't matter. Tried playing with my friend on the LAN server with mods. He has around 30-40 fps with shaders with his i5-7400 + gtx1060 while I'm having 10-20 on gtx1080ti... I jumped to 1.16.x from 1.12.2 - there was no such problem.

**Comment by ajbsddfkj** [comment]
> Might be helpful for someone:
> 
> 1. **Make sure Minecraft (Java) is using your discrete GPU**, not integrated. you can check this by opening `Shaders` in options and check on the bottom of the screen:
>    ![image](https://user-images.githubusercontent.com/32597776/93719235-5444dd00-fbab-11ea-801c-5702062fd7d5.png)
>    If it says your discrete GPU (Nvidia / AMD), that means it's already using your discrete GPU.
>    If it's using integrated GPU (Intel HD or AMD Vega), change your GPU settings, follow [this guide](https://windowsreport.com/minecraft-not-using-gpu/#:~:text=Click%20the%20Switchable%20Graphics%20tab,option%20for%20the%20discrete%20GPU.) to change what GPU to use in Minecraft (guide for both AMD and Nvidia card are there)
> 2. **Reduce render distance**, I found that some shaders are poorly optimized when you have a high render distance (16+), some that I know are BSL and Sildurs, I usually use 8-12 render distance with BSL / Sildurs, more than that I will start getting FPS drop especially when I already rendered all of the chunks within the render distance area. Some shaders doesn't impact too much performance when using high render distance, even on 48 chunks, like SEUS Renewed.
>    For performance reference (I'm using Ryzen 3600 & GTX 1080ti on 1440p monitor):
>    
>    * **BSL v7.2**
>      8 chunks = 90-100 fps
>      16 chunks = 80-90 fps ( **-10%** from 8 )
>      32 chunks = 55-65 fps ( **-37%** from 8, **-29%** from 16)
>    * **Sildurs v1.262 Extreme VL**
>      8 chunks = 115-120 fps
>      16 chunks = 104-110 fps ( **-8.5%** from 8 )
>      32 chunks = 65-70 fps ( **-42%** from 8, **-36.5%** from 16)
>    * **Seus Renewed v1.0.1**
>      8 chunks = 100-105 fps
>      16 chunks = 90-95 fps ( **-9.8%** from 8 )
>      32 chunks = 89-92 fps ( **-11.8%** from 8, **-3.23%** from 16)

Suspiciouslookingowl you saved my ass, I only created github to tell you that you are a fcking genius <3 <3 <3 <3. Seriously game is 50 times better. Thx a lot bro

**Comment by yermolim** [comment]
> > Same thing. Have around 10-20 fps with any shaders and any shader presets (from low to ultra), any resolution (from tiny window to 4k) and any draw distance on my i5-8400 + gtx1080ti. GPU load is around 30-70% depending on settings. Game version is 1.16.3. Optifine version is G5.
> 
> Tried again without any mods (only OF with default settings) - same thing around 20 fps with shaders regardless of settings. TPS is stable. Lagometer is mostly white. CPU load is around 30% (none of the cores hits 100%), GPU load is around 35%. Single or multiplayer - doesn't matter. Tried playing with my friend on the LAN server with mods. He has around 30-40 fps with shaders with his i5-7400 + gtx1060 while I'm having 10-20 on gtx1080ti... I jumped to 1.16.x from 1.12.2 - there was no such problem.

Ok, looks like it's a [minecraft render engine issue since 1.15.x](https://bugs.mojang.com/browse/MC-164123) which still persists even in pre-1.17. And things become even worse with optifine+shaders on. The only thing I can't understand if it's a render engine issue then why it's not affecting everyone...

**Comment by jsauve** [comment]
> Might be helpful for someone:
> 
> 1. **Make sure Minecraft (Java) is using your discrete GPU**, not integrated. you can check this by opening `Shaders` in options and check on the bottom of the screen:
>    ![image](https://user-images.githubusercontent.com/32597776/93719235-5444dd00-fbab-11ea-801c-5702062fd7d5.png)
>    If it says your discrete GPU (Nvidia / AMD), that means it's already using your discrete GPU.
>    If it's using integrated GPU (Intel HD or AMD Vega), change your GPU settings, follow [this guide](https://windowsreport.com/minecraft-not-using-gpu/#:~:text=Click%20the%20Switchable%20Graphics%20tab,option%20for%20the%20discrete%20GPU.) to change what GPU to use in Minecraft (guide for both AMD and Nvidia card are there)
> 2. **Reduce render distance**, I found that some shaders are poorly optimized when you have a high render distance (16+), some that I know are BSL and Sildurs, I usually use 8-12 render distance with BSL / Sildurs, more than that I will start getting FPS drop especially when I already rendered all of the chunks within the render distance area. Some shaders doesn't impact too much performance when using high render distance, even on 48 chunks, like SEUS Renewed.
>    For performance reference (I'm using Ryzen 3600 & GTX 1080ti on 1440p monitor):
>    
>    * **BSL v7.2**
>      8 chunks = 90-100 fps
>      16 chunks = 80-90 fps ( **-10%** from 8 )
>      32 chunks = 55-65 fps ( **-37%** from 8, **-29%** from 16)
>    * **Sildurs v1.262 Extreme VL**
>      8 chunks = 115-120 fps
>      16 chunks = 104-110 fps ( **-8.5%** from 8 )
>      32 chunks = 65-70 fps ( **-42%** from 8, **-36.5%** from 16)
>    * **Seus Renewed v1.0.1**
>      8 chunks = 100-105 fps
>      16 chunks = 90-95 fps ( **-9.8%** from 8 )
>      32 chunks = 89-92 fps ( **-11.8%** from 8, **-3.23%** from 16)

SUCCESS! After looking at many other posts and forums, this is what led me to the answer that finally made shaders perform the way they're supposed to on a GTX 1060 (in an Acer Predator Helios 300 laptop). You need to tell Windows to use the Nvidia GPU for `javaw.exe`. Now we're getting a nice ~60fps for shaders at a chunk setting of around 10.

**Comment by CxncxerXD** [comment]
my quality looked realy cheep but if u go to ur shaders u can put render quality all the way up to 2 and it looks better.
u can also mess with the settings to make it look better for urself.¨
hope it helped:)

**Comment by Bighelp82** [comment]
Hi all, Ive never posted or used this site much, but I decided to come here after literally days of desperation, cleaning drivers, redownloading, Trying over and over to figure out why my 2080 TI and I9 9900k cannot run minecraft with 32 render distance and shaders without dipping below 30, and The solution was posted above, I can confirm that any minecraft install after 1.14.4 will more or less break your game if you try to use shaders, destroying fps, now I can keep a smooth 60 after not using 1.16.5, and the way my world is generating seems 100 times smoother. Yermolim really saved me alot of time and stress as ive been trying to figure this out for days, if your minecraft is running horribly with shaders, revert to any install below 1.15.

**Comment by kevorino** [comment]
Yep ^ 
I was the guy posting updates on upgrading components on my pc to find out if it was my pc lagging/ crashing the game.
After getting my hands on the 3080 the game no longer crashed.. that was a specific issue with my old 1080ti.
However, the game still lagged on versions 1.15+ considerably more than older versions when shaders were enabled and no matter what settings within the shaders. (High, Low, Extreme , Ultra) The only option that increased FPS above 40-50 ish was render distance below or at 4.

Conclusion, lag issue definitely still exists.

This is my previous post incase anyone wants to check them out:
https://github.com/sp614x/optifine/issues/4288#issuecomment-670550469

**Comment by Bighelp82** [comment]
As of Now on 14.4 i can keep a solid 60 fps with 32 render distance and shaders on ultra settings. Idk whats up with the newer versions of minecraft.

**Comment by ZodTheTod** [comment]
for anyone who cant seem to fix this problem by using any of the above methods, check if you have anisotropic filtering turned on. if it is turned to anything above off then turn it off. worked for me.

**Comment by kevorino** [comment]
> for anyone who cant seem to fix this problem by using any of the above methods, check if you have anisotropic filtering turned on. if it is turned to anything above off then turn it off. worked for me.

Yeah, that's definitely turned off for me and I still get lag with shaders. It is also weird that this setting makes me lag with it enabled..

Possibly something that helps:
Complementary Shaders POTATO Mode and Builders QOL Shaders are the only shaders that I noticed don't get affected by the FPS drops.
Main difference is that they do not contain Shadows, it's disabled and only contain effects like water shaders, bloom and light ray thingies.

**Comment by xaLi112** [comment]
Same problem i have a RTX 3060, and i can't even run stable 30fps with shaders. could not find any solution.

**Comment by Vision1x** [comment]
It looks like this is something that the newer versions in minecraft have a problem with the 3000 series. I have a 3070 myself and are struggling sitting around 60-70 fps with Seus Renewed Shaders in 1.17.1. But as i was experimenting i tried out shaders in 1.14.2 and i got instantly 180-240 fps. Im trying to find out where the problem is but no luck yet

**Comment by XenoxYT** [comment]
Try with 1.16.5

**Comment by Swiezu** [comment]
> 
> 
> It looks like this is something that the newer versions in minecraft have a problem with the 3000 series. I have a 3070 myself and are struggling sitting around 60-70 fps with Seus Renewed Shaders in 1.17.1. But as i was experimenting i tried out shaders in 1.14.2 and i got instantly 180-240 fps. Im trying to find out where the problem is but no luck yet

I'm agree with that. I have same issue, sometimes 60fps with drops to 30 on seus renewed 1.0.1 (mc 1.17.1) (3080 TI suprim x).

**Comment by GreySux** [comment]
One of the things that isn't being talked about here enough: Anisotropic Filtering quality. This tanks so much FPS.

Options>Video settings>Quality>Anisotropic Filtering 

Personally I go without it, doesn't change quality very much and makes game run much more smoothly.

**Comment by Collin146** [comment]
So i recently upgraded my hardware (RTX 3060, Ryzen 5 5600x, 32GB RAM) and i still had major FPS loss whilst using shaders. I tried countless options and still couldn't figure out what was causing the issues. 

However, trying countless things, i found out that **Threaded Optimization** in the NVIDIA Control Panel (which is enabled by default) causes major lag when using shaders. When turning it off for the application, i must've jumped from 20- 30 FPS in certain areas to 100 - 120. I hope this helps anyone.

**Comment by TheKuyPeD** [comment]
> So i recently upgraded my hardware (RTX 3060, Ryzen 5 5600x, 32GB RAM) and i still had major FPS loss whilst using shaders. I tried countless options and still couldn't figure out what was causing the issues.
> 
> However, after trying countless things, i found out that **Threaded Optimization** in the NVIDIA Control Panel (which is enabled by default) causes major lag when using shaders. When turning it off for the application, i must've jumped from 20- 30 FPS in certain areas to 100 - 120. I hope this helps anyone.

I turned it off so FPS increase from 30-45 to 60. Thank you so much for method
